core/src/com/unciv/ui/newgamescreen/PlayerPickerTable.kt

```diff
@@ -28,12 +28,20 @@ class PlayerPickerTable(val newGameScreen: NewGameScreen, val newGameParameters:
         update()
     }
 
-    fun update() {
+    fun update(desiredCiv: String = "") {
         playerListTable.clear()
         val gameBasics = newGameScreen.ruleset // the mod picking changes this ruleset
+        var desiredCivPresent = false
+        var anyRandomPlayer: Player? = null
         for (player in newGameParameters.players) {
             if(!newGameScreen.ruleset.nations.containsKey(player.chosenCiv)) // this was in a mod we disabled
                 player.chosenCiv="Random"
+            if (player.chosenCiv==desiredCiv) desiredCivPresent = true
+            if (player.chosenCiv=="Random" && anyRandomPlayer==null && player.playerType==PlayerType.Human) anyRandomPlayer = player
+        }
+        if (desiredCiv.isNotEmpty() && !desiredCivPresent && anyRandomPlayer!=null)
+            anyRandomPlayer.chosenCiv = desiredCiv
```

Comment:
All of this should be moved to a separate function called "addDesiredCiv" or something

Reply:
You mean leave update() as it was, no parameters, and have two calls in the lambda in NewGameScreen line 33? Seems ugly.

Anyway, I can't push anything, Android studio suddenly marked the branch as having outstanding commits, and those are a jumble of commits for other branches, and I have no idea how this chaos came to be. It even says I merged the mod loader branch into this, but I definitely didn't! 

I'll have to delete everything and start over. Again. Or move on.

Reply:
????
Everything looks fine from here
No extraneous changes, no conflicts with main branch...

Reply:
And, no, I meant 'make the extra function, and call it from this function' =)

Reply:
> ????
> Everything looks fine from here
> No extraneous changes, no conflicts with main branch...

Of course - Because I didn't push when I saw the push window. But if I pushed now - I fear a big 'oops'

Reply:
I think I understand where I went wrong - I again expected TFS rules to apply where a branch *is* an independent copy of all files (except it's shadow copies under the hood but that doesn't affect you). Merging branches doesn't affect the source branch unless you allow it to delete the source after merge (which it does too). Git just reassigns commits, and if you delete a branch carrying commits it seems to reassign those into any random branch happening to be floundering around - or at least it looks like it.